### PR TITLE
feat(smolagents): Populate LLM Provider & System Attributes

### LINKED ISSUE #2615: [feature request] Smolagents - Populate LLM Provider & System Attributes
**Is your feature request related to a problem? Please describe.**
NA

**Describe the solution you'd like**
The `model_id` from `LiteLLMModel` or `LiteLLMRouterModel` can be used to infer `llm.system` based on model patterns. For `llm.provider`, we can first check the `api_base` URL or the specific SDK class used to call the model.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by ialisaleh**
Closes #2615 

---

> [!NOTE]
> **Low Risk**
> Low risk instrumentation change that only adds optional `LLM_PROVIDER`/`LLM_SYSTEM` attributes to LLM spans; main risk is misclassification due to heuristic inference from model IDs/endpoints.
> 
> **Overview**
> Smolagents LLM spans now populate **`LLM_PROVIDER`** and **`LLM_SYSTEM`** attributes by inferring values from the model class name, configured endpoint URL, and `model_id` prefixes.
> 
> Tests are updated to assert these new attributes for OpenAI and Anthropic models, and new unit tests cover the provider/system inference helpers across supported providers/endpoints and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5edbbb1884fee72d06f1067e6bd4292d690c6c05. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by caroger** [comment]
@claude please review this PR

File: python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py
Comment (axiomofjoy) [inline python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py:1]: I think we could use some unit tests on the `infer_llm_provider_from_class_name` and `infer_llm_provider_from_endpoint` helpers to cover those different conditionals.

File: python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py [hunk 1]
```diff
@@ -536,12 +546,151 @@ def _has_active_llm_parent_span() -> bool:
...
+    if host.endswith("x.ai"):
+        return OpenInferenceLLMProviderValues.XAI
+
+    if host.endswith("deepseek.com"):
+        return OpenInferenceLLMProviderValues.DEEPSEEK
+
+    return None
+
+
+def infer_llm_system_from_model(
```
Comment (axiomofjoy) [inline python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py:641]: Is there a registry of model names in litellm (importable constants) we could use for unit tests?

File: python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py
Comment (ialisaleh) [reply python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py:1]: Hi @axiomofjoy - I updated the PR to add the required testcases (`TestInferLLMProviderFromClassName` / `TestInferLLMProviderFromEndpoint` / `TestInferLLMSystemFromModel`) for the LLM provider/system helpers to cover all the conditionals. Please take a look. Thanks!

File: python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py (hunk 1 above)
Comment (ialisaleh) [reply python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py:641]: Hi @axiomofjoy - I used multiple model names from different LLMs to test all the conditionals. Thanks!

File: python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py [hunk 2]
```diff
@@ -536,12 +546,151 @@ def _has_active_llm_parent_span() -> bool:
...
+            "davinci",
+            "curie",
+            "babbage",
+            "ada",
+            "azure_openai",
+            "azure_ai",
+            "azure",
+        )
+    ):
+        return OpenInferenceLLMSystemValues.OPENAI
```
[the remark this answers is not in the document]
Comment (ialisaleh) [reply python/instrumentation/openinference-instrumentation-smolagents/src/openinference/instrumentation/smolagents/_wrappers.py:667]: We use the `model_id` value in the `infer_llm_system_from_model()` method, which does not include the provider like LiteLLM prefix, so we don't need to add the `OPENAI `provider prefix check.
